azext_iot/common/base_discovery.py

```diff
@@ -0,0 +1,345 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from abc import abstractmethod
+from azure.core.exceptions import HttpResponseError
+from knack.util import CLIError
+from knack.log import get_logger
+from azext_iot.common.shared import AuthenticationTypeDataplane
+from typing import Dict, List
+from types import SimpleNamespace
+
+logger = get_logger(__name__)
+
+
+# Abstract base class
+class BaseDiscovery(object):
```

Comment:
For the class to be abstract it should inherit from abc.ABC, otherwise it can be directly instantiated.